autogpt/core/plugin/base.py

```diff
@@ -1,44 +1,95 @@
 import abc
 import dataclasses
-import typing
-from enum import Enum
+from enum import StrEnum
+from typing import Type
 
-if typing.TYPE_CHECKING:
-    from autogpt.core.command import Command
-    from autogpt.core.configuration import Configuration
-    from autogpt.core.logging import Logger
-    from autogpt.core.workspace import Workspace
+from autogpt.core.budget import BudgetManager
+from autogpt.core.command import Command, CommandRegistry
+from autogpt.core.llm import LanguageModel
+from autogpt.core.memory import MemoryBackend
+from autogpt.core.planning import Planner
 
+# Expand to other types as needed
+PluginType = (
+    Type[BudgetManager]
+    | Type[Command]
+    | Type[CommandRegistry]
+    | Type[LanguageModel]
+    | Type[MemoryBackend]
+    | Type[Planner]
+)
 
-class PluginType(Enum):
-    COMMAND = 0
-    OPENAPI = 1
-    SYSTEM_REPLACEMENT = 2
-    MESSAGE_SINK = 3
-    LOGGING_SINK = 4
+Plugin = (
+    BudgetManager | Command | CommandRegistry | LanguageModel | MemoryBackend | Planner
+)
+
+
+class PluginStorageFormat(StrEnum):
+    """Supported plugin storage formats."""
+
+    AUTO = "auto"  # We'll try to determine the plugin load location
+    AUTOGPT_PLUGIN_REPO = "autogpt_plugin_repo"  # Grab them from our managed repo
+    WORKSPACE = "workspace"  # Grab them from the workspace
+    OTHER_FILE_PATH = "other_file_path"  # Grab them from a file path
+    INSTALLED_PACKAGE = "installed_package"  # Grab them from an installed package
+    # PYPI = "pypi"  # Grab them from pypi
+    # GITHUB = "github"  # Grab them from a github repo
```

Comment:
Shouldn't limit to git
```suggestion
    # GIT = "git"  # Grab them from a git repo
```